### DISCUSSION TITLE
fr/models/

### DISCUSSION THREAD

**Discussion by giscus** [ultralytics/ultralytics#9323] [Docs]
# fr/models/

Explore la gamme variée de modèles de la famille YOLO , SAM, MobileSAM, FastSAM, YOLO-NAS, YOLO-World et RT-DETR pris en charge par Ultralytics. Commence par des exemples d'utilisation de CLI et Python .

https://docs.ultralytics.com/fr/models/

**Comment by Hamzaboughfir123** [comment]
I would like to combine two yolov8 models to perform as one, there is way to do that?

**Reply by pderrenger** [reply]
Hey there! Combining two YOLOv8 models into a single one would require an ensemble approach where predictions from both models are merged in some fashion. While we don't have a built-in feature for this in Ultralytics YOLO, you can manually implement it by running inference with both models individually and then combining their results. 🤖✨

Here's a simple conceptual example in Python:

```python
from ultralytics import YOLO

# Load your models
model1 = YOLO('path/to/your/first_model.pt')
model2 = YOLO('path/to/your/second_model.pt')

# Run inference separately
results1 = model1('path/to/your/image.jpg')
results2 = model2('path/to/your/image.jpg')

# Now you can combine results1 and results2 in a way that suits your needs
# This could involve averaging their predictions, taking the union of their detections, etc.
```

Keep in mind, the method to combine the models' outputs depends heavily on your specific use case! 😊 Hope this helps!

**Reply by Hamzaboughfir123** [reply]
yes but how to combine model 1 and model 2 to have just one result in inference knowing that i have the same classes for the 2 models

**Reply by Hamzaboughfir123** [reply]
and how to combine the results

**Reply by pderrenger** [reply]
@Hamzaboughfir123 combining results from two models with the same classes involves aggregating their detections. A simple approach is to average their confidence scores and run Non-Maximum Suppression (NMS) across all detections to eliminate overlaps. This assumes both models predict the same classes and you wish to combine their strengths.

Here's a conceptual snippet in Python:

```python
from ultralytics import YOLO, non_maximum_suppression as nms

# Load your models
model1, model2 = YOLO('path/to/first_model.pt'), YOLO('path/to/second_model.pt')

# Inference
img = 'path/to/image.jpg'
results1, results2 = model1(img), model2(img)

# Combine detections (this is a simplified example)
detections = torch.cat((results1.pred[0], results2.pred[0]), 0)  # Concatenate detections

# Apply NMS
keep = nms(detections, iou_threshold=0.5, conf_threshold=0.5)
combined_results = detections[keep]

# Display or process combined_results
```

This is a simplified way to ensemble two models. Adjust according to your needs, especially handling different confidence scores and employing appropriate thresholds. 🚀

**Comment by Zelyanne** [comment]
Hello please, is there a way for making yolov11 to predict 2 class per detection so that when it fails to detect the first class , if the second class is detected it ll show it

**Reply by glenn-jocher** [reply]
@Zelyanne thank you for your question. YOLO models are designed to predict a single class per object detection, but multi-class labeling per detection isn't natively supported. You could implement a custom post-processing step to assign secondary classes based on your criteria or train the model with augmented multi-label datasets. For advanced use cases like this, consider modifying the model or leveraging additional resources to meet your requirements.
